appcues/src/main/java/com/appcues/analytics/Analytics.kt

```diff
@@ -0,0 +1,188 @@
+
+internal interface QualificationService {
+
+    // will merge and hit the backend to see if it qualifies, returning QualificationResult
+    suspend fun qualify(intents: List<AnalyticsIntent>): QualificationResult?
+}
+
+internal interface RenderingService {
+
+    // will attempt to show experiences by order
+    // clearCache is mostly used when we know the qualification is from a screen_view
+    suspend fun show(experiences: List<Experience>, clearCache: Boolean)
+
+    suspend fun reset()
+}
+
+internal interface SessionService {
+
+    // returns whether session is started or not
+    suspend fun isSessionStarted(): Boolean
+
+    // starts session (can start using incoming intent or existing stored information
+    suspend fun startSession(intent: AnalyticsIntent): Boolean
+
+    // decorates incoming intent and returns a new intent with session/auto props information
+    suspend fun decorateIntent(intent: AnalyticsIntent): AnalyticsIntent
+
+    suspend fun reset()
+}
+
+// new model for Intents available for Analytics
+internal sealed class AnalyticsIntent(open val isInternal: Boolean, val timestamp: Date = Date()) {
+
+    object Anonymous : AnalyticsIntent(false)
+
+    data class Identify(
+        val userId: String,
+        val properties: Map<String, Any>?
+    ) : AnalyticsIntent(false)
+
+    data class UpdateProfile(
+        val properties: Map<String, Any>?,
+        override val isInternal: Boolean
+    ) : AnalyticsIntent(isInternal)
+
+    data class UpdateGroup(
+        val groupId: String?,
+        val properties: Map<String, Any>?,
+        override val isInternal: Boolean
+    ) : AnalyticsIntent(isInternal)
+
+    data class Event(
+        val name: String,
+        val attributes: Map<String, Any>?,
+        val context: Map<String, Any>?,
+        override val isInternal: Boolean
+    ) : AnalyticsIntent(isInternal)
+}
+
+internal class Analytics(
+    private val appcuesCoroutineScope: AppcuesCoroutineScope,
+    private val intentQueue: AnalyticsIntentQueue,
+    private val qualificationService: QualificationService,
+    private val renderingService: RenderingService,
+    private val sessionService: SessionService,
+) : IntentProcessor, ApplicationMonitor.Listener {
+
+    companion object {
+
+        const val EVENT_ATTR_SCREEN_TITLE = "screenTitle"
+        const val EVENT_CONTEXT_SCREEN_TITLE = "screen_title"
+    }
+
+    init {
+        intentQueue.setProcessor(this)
+
+        ApplicationMonitor.subscribe(this)
+    }
+
+    private val _analyticsFlow = MutableSharedFlow<AnalyticsIntent>(1)
+    val analyticsFlow: SharedFlow<AnalyticsIntent>
+        get() = _analyticsFlow
+
+    fun anonymous() {
+        intentQueue.flushAndProcess(Anonymous)
+    }
+
+    fun identify(userId: String, properties: Map<String, Any>? = null) {
+        intentQueue.flushAndProcess(Identify(userId, properties))
+    }
+
+    fun updateProfile(properties: Map<String, Any>? = null, isInternal: Boolean) {
```

Comment:
adding updateProfile is an idea that make explicitly for places like: UpdateProfileAction we would need to add it to the API.

Reply:
Would not need in the public API right? the UpdateProfileAction execute is already a suspend fun, so it could get the user ID from storage just like it does now, but safely?

Reply:
yes, I think it could also import Analytics and call this directly.

Reply:
this is a side idea, that maybe this could be beneficial for the customer to have this option to update the current user profile without the needing to store the current user somewhere else. good utility method maybe

Reply:
perhaps, I'd rather not expand the public API though unless there is very clear demand for it. I'd assume the customer app already knows and stores the user information somewhere for it's own purposes so probably not a big deal. And maintains consistency with web and all the other mobile platforms to keep our identify call behavior as-is.